sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -0,0 +1,137 @@
+// SPDX-FileCopyrightText: 2021 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.scl.ied;
+
+
+import org.apache.commons.lang3.StringUtils;
+import org.lfenergy.compas.scl2007b4.model.TCompasFlow;
+import org.lfenergy.compas.scl2007b4.model.TExtRef;
+import org.lfenergy.compas.scl2007b4.model.TInputs;
+import org.lfenergy.compas.sct.commons.dto.SclReport;
+import org.lfenergy.compas.sct.commons.scl.PrivateService;
+import org.lfenergy.compas.sct.commons.scl.SclElementAdapter;
+import org.lfenergy.compas.sct.commons.util.Utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A representation of the model object
+ * <em><b>{@link InputsAdapter InputsAdapter}</b></em>.
+ *
+ * @see LN0Adapter
+ * @see AbstractLNAdapter
+ * @See TInputs
+ * @See TExtRef
+ */
+public class InputsAdapter extends SclElementAdapter<LN0Adapter, TInputs> {
+
+    /**
+     * Constructor
+     *
+     * @param parentAdapter Parent container reference
+     * @param tInputs       Current reference
+     */
+    public InputsAdapter(LN0Adapter parentAdapter, TInputs tInputs) {
+        super(parentAdapter, tInputs);
+    }
+
+    /**
+     * Check if current element is a child of the parent element
+     *
+     * @return true if the currentElem is part of the parentAdapter children
+     */
+    @Override
+    protected boolean amChildElementRef() {
+        return currentElem == parentAdapter.getCurrentElem().getInputs();
+    }
+
+    @Override
+    protected String elementXPath() {
+        return "Inputs";
+    }
+
+    /**
+     * Update iedName of ExtRefs in this "Inputs" element.
+     * @return list of encountered errors
+     */
+    public List<SclReport.ErrorDescription> setExtRefNames() {
```

Comment:
Humm A setter returning a list of errors ? why not throwing them ?

Reply:
It is not really a setter (in the sense of usual getter/setter). Maybe the name can be changed to `updateExtRefNames()`.
If I throw an Exception, I have to catch it in  `ExtRefService#setExtRefIedNames` to create the `SclReport` object. I think it would be a bad usage to use Exception. But I am open to discussion.